From a11f67fd820c9a9693f7df8a11c3533cbd6b8ee2 Mon Sep 17 00:00:00 2001
From: Joris Vaillant <joris.vaillant@inria.fr>
Date: Tue, 4 Jun 2024 17:09:40 +0200
Subject: [PATCH 1/2] cmake: Turn `BUILD_WITH_LIBPYTHON` OFF when building with
 PyPy

This already had been fixed in 4d521ec but had been lost in CMakeLists
refactoring.
---
 CMakeLists.txt | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 048a03b67..99d06a1cf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -280,15 +280,17 @@ if(BUILD_PYTHON_INTERFACE)
 
   message(STATUS "Python compiler: ${_python_implementation_value}")
   if(_python_implementation_value MATCHES "PyPy")
-    set(BUILD_PYTHON_INTERFACE_WITH_PYPY TRUE)
+    set(BUILD_WITH_LIBPYTHON OFF)
+    message(
+      STATUS
+        "PyPy detected, therefore libpython is not available and BUILD_WITH_LIBPYTHON set to OFF.")
   endif()
-
-else(BUILD_PYTHON_INTERFACE)
+else()
   message(
     STATUS
       "Pinocchio won't be compiled with its Python bindings. If you want to enable this feature, please set the option BUILD_PYTHON_INTERFACE to ON."
   )
-endif(BUILD_PYTHON_INTERFACE)
+endif()
 
 if(BUILD_WITH_HPP_FCL_SUPPORT)
   list(APPEND CFLAGS_DEPENDENCIES "-DPINOCCHIO_WITH_HPP_FCL")
-- 
2.34.1

